### PR TITLE
Build container: Base build image on Debian Stretch

### PR DESCRIPTION
**Pull request by aknuds1**
**What this PR does / why we need it**:
Base Grafana build image on Debian Stretch (LTS into 2022), _or_ on Ubuntu 18.04 (also LTS into 2022). My personal preference is Debian, but I'd like for us to discuss which one to go with. Both seem to produce binaries with the same compatibility profile (at least binaries from both work with Ubuntu 18.04 ARM64).

This PR also simplifies the [FPM](https://fpm.readthedocs.io/) installation, since the original approach didn't work on Debian and I didn't see why we can't use the Ruby installed via `apt-get` (instead of via `rvm`).

I've tested the binaries built with this image, and they are compatible with Ubuntu 18.04 ARM64, so should be a good choice. The current stable Debian, Buster, unfortunately produces incompatible binaries (as Ubuntu 20.04 does).

I've also tested binaries built in a container based on Ubuntu 18.04, these seem to have the same compatibility profile (i.e. linking to glibc 2.27) as the ones built with Debian Stretch (and compatible with Ubuntu 18.04 ARM64).

@RichiH Do you have anything to say with regards to whether we should use Debian Stretch over Ubuntu 18.04 for building our images or vice versa? I know @Estrax already gave his vote to Debian Stretch.

**Which issue(s) this PR fixes**:
ARM64 binaries built with our current build image aren't compatible with Ubuntu 18.04 (ARM64). See #24844.

### CODE REVIEW

File: scripts/build/ci-build/Dockerfile [hunk 1]
```diff
@@ -130,9 +131,6 @@ RUN cd /tmp && \
     tar xf x86_64-linux-musl-cross.tgz && \
     rm x86_64-linux-musl-cross.tgz
 
-RUN apt-get install -yq gcc libc-dev make && \
-    gpg2 --keyserver hkp://keys.gnupg.net --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3 7D2BAF1CF37B13E2069D6956105BD0E739499BDB && \
-    curl -fsSL https://get.rvm.io | bash -s stable && \
-    /bin/bash -l -c "rvm requirements && rvm install 2.2 && gem install -N fpm"
+RUN apt-get install -yq ruby ruby-dev rubygems && gem install -N fpm
```
Comment (bergquist) [inline scripts/build/ci-build/Dockerfile:134]: perhaps we should move this to line 86 as well?

**Review by bergquist — COMMENTED** [review]
I dont have strong opinions on this. But I would favor Debian over Ubuntu.

I've probably said it before but the main reason we switched to Ubuntu was that our docker images on dockerhub had too many security vulnerabilities and the debian images didn't offer a good solution out of that (2 years ago maybe) while ubuntu did.

**Comment by aknuds1** [comment]
> I dont have strong opinions on this. But I would favor Debian over Ubuntu.
> 
> I've probably said it before but the main reason we switched to Ubuntu was that our docker images on dockerhub had too many security vulnerabilities and the debian images didn't offer a good solution out of that (2 years ago maybe) while ubuntu did.

@bergquist Thanks for providing that historical context, wasn't aware of it :)

**Comment by bergquist** [comment]
@xlson I'd love to hear your opinion on this. It sounds like a good idea to me but you know more on this subject.

**Comment by xlson** [comment]
I honestly don't have a strong opinion. To me, the priority is stability, if Debian gives us more solid ground to stand on I think we should go with it.

Great work on digging into these problems @aknuds1.

**Review by xlson — APPROVED** [review]

**Comment by aknuds1** [comment]
Thanks @xlson! Any idea for what release we should start using the new build image (and fixing binary compatibility)? 7.0.2?

**Comment by RichiH** [comment]
As Email pointed out, we can now drop this as Stretch LTS is EOL. I don't think we need to care about ELTS.

**Pull request merged**